host/matching_simulation_test.go

```diff
@@ -302,36 +299,43 @@ func (s *MatchingSimulationSuite) generate(
 				}
 				return
 			}
-			scheduleID := int(atomic.AddInt32(lastTaskScheduleID, 1))
-			if scheduleID > maxTasksToGenerate {
-				s.log("Generated %d tasks so generator will stop", maxTasksToGenerate)
+			newTasksGenerated := int(atomic.AddInt32(tasksGenerated, 1))
+			if newTasksGenerated > taskConfig.getMaxTasksToGenerate() {
+				s.log("Generated %d tasks so generator will stop", newTasksGenerated)
 				return
 			}
-			decisionTask := newDecisionTask(domainID, tasklist, scheduleID)
+			isolationGroup := ""
+			if len(taskConfig.getIsolationGroups()) > 0 {
+				isolationGroup = taskConfig.getIsolationGroups()[newTasksGenerated%len(taskConfig.getIsolationGroups())]
```

Comment:
This is a good happy path scenario, but you'll want to control this as a config parameter for more tricky 'happy-path' scenarios. 

expected scenarios include things like: 
- there being no pollers in a zone, therefore we should see the isolation group logic re-route traffic 
- a poller was in a zone with a sticky tasklist and then went away
- The isolation-group is being flooded due to skew

as well as pathological scenarios like:
- traffic is dispatched to an isolation-group with too many partitions (eg partition > pollers per IG)